src/R/Editor/Impl/Data/WorkspaceVariableProvider.cs

```diff
@@ -131,35 +132,29 @@ internal sealed class WorkspaceVariableProvider : RSessionChangeWatcher, IVariab
-                    if (debugSessionProvider != null) {
-                        var debugSession = await debugSessionProvider.GetDebugSessionAsync(Session);
-                        if (debugSession != null) {
-                            var stackFrames = await debugSession.GetStackFramesAsync();
-
-                            var globalStackFrame = stackFrames.FirstOrDefault(s => s.IsGlobal);
-                            if (globalStackFrame != null) {
-                                const DebugEvaluationResultFields fields =
-                                    DebugEvaluationResultFields.Expression |
-                                    DebugEvaluationResultFields.Kind |
-                                    DebugEvaluationResultFields.TypeName |
-                                    DebugEvaluationResultFields.Classes |
-                                    DebugEvaluationResultFields.Length |
-                                    DebugEvaluationResultFields.SlotCount |
-                                    DebugEvaluationResultFields.AttrCount |
-                                    DebugEvaluationResultFields.Dim |
-                                    DebugEvaluationResultFields.Flags;
-                                DebugEvaluationResult evaluation = await globalStackFrame.EvaluateAsync("base::environment()", "Global Environment", fields, "rtvs:::make_repr_str()");
-                                var e = new RSessionDataObject(evaluation);  // root level doesn't truncate children and return every variables
-
-                                _topLevelVariables.Clear();
-
-                                var children = await e.GetChildrenAsync();
-                                if (children != null) {
-                                    foreach (var x in children) {
-                                        _topLevelVariables[x.Name] = x; // TODO: BUGBUG: this doesn't address removed variables
-                                    }
-                                }
+                    var stackFrames = await session.TracebackAsync();
+
+                    var globalStackFrame = stackFrames.FirstOrDefault(s => s.IsGlobal);
+                    if (globalStackFrame != null) {
+                        const REvaluationResultProperties fields =
+                            REvaluationResultProperties.ExpressionProperty |
+                            REvaluationResultProperties.AccessorKindProperty |
+                            REvaluationResultProperties.TypeNameProperty |
+                            REvaluationResultProperties.ClassesProperty |
+                            REvaluationResultProperties.LengthProperty |
+                            REvaluationResultProperties.SlotCountProperty |
+                            REvaluationResultProperties.AttributeCountProperty |
+                            REvaluationResultProperties.DimProperty |
+                            REvaluationResultProperties.FlagsProperty;
```

Comment:
Nope, just didn't finish going over all these before committing by mistake. I'll do `using static` in all places where there's more than a couple.